wrappers/Julia/CoolProp.jl

```diff
@@ -1,190 +1,677 @@
-VERSION < v"0.4.0" && __precompile__()
```

Comment:
The version is only relevant because the `__precompile__()` did not exist before version 4.
So if their package requires Julia 4 or 5, they simply do not need to check before as it is already done in the requirements.

I see that in the repository to make the Julia package, we have `julia 0.4` as requirement. Thus, if we keep this requirement, we do not need the other checking. I'm however wondering if we shouldn't lower this to `julia 0.3` as it should still be compatible with thanks to `Compat` and to specify the tag in the same way as [Compat](https://github.com/JuliaLang/Compat.jl/blob/master/src/Compat.jl#L1), so having the version limit.
Before this update, the wrapper was still compatible with `julia 0.3`, but I have no idea if there are any users of CoolProp still using Julia 0.3. So I don't know if we should keep compatibility with (it just look to me it is a no pain move thanks to Compat).

This do not need to be on the development and could be only on the release. However, I think it can be dangerous as we would not see issues with top level statements being executed at compile time and no more at run time. Having it during the development also allows to debug all issues before making a release.